src/NUnitFramework/framework/Attributes/OSPlatformTranslator.cs

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace NUnit.Framework
+{
+    /// <summary>
+    /// Marks an assembly, test fixture or test method as applying to a specific platform.
+    /// </summary>
+    internal static class OSPlatformTranslator
+    {
+        /// <summary>
+        /// Converts one or more .NET 5+ OSPlatformAttributes into a single NUnit PlatformAttribute
+        /// </summary>
+        /// <param name="allAttributes">Enumeration of all attributes.</param>
+        /// <returns>All attributes with OSPlatformAttributes translated into PlatformAttributes.</returns>
+        public static IEnumerable<object> Translate(IEnumerable<object> allAttributes)
+        {
+            var includes = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            var excludes = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var attribute in allAttributes)
+            {
+                if (attribute is PlatformAttribute platformAttribute)
+                {
+                    Add(includes, platformAttribute.Include);
```

Comment:
`HashSet<T>` defines a `UnionWith` function to merge another `IEnumerable<T>` to itself. Would it work to use that here?

Reply:
Replaced own loop with call to `UnionWith`.